src/createElement.js

```diff
@@ -0,0 +1,36 @@
+export function createElement(type, props, ...children) {
+  props = props || {};
+
+  // 1. 중첩된 배열 평탄화
+  let flatChildren = children.flat(Infinity);
+
+  // 2. falsy 값(null, undefined, boolean) 제거
+  flatChildren = flatChildren.filter(
+    (child) => child != null && typeof child !== "boolean"
+  );
+
+  // 3. 문자열/숫자 children을 TEXT_ELEMENT로 변환
+  flatChildren = flatChildren.map((child) => {
+    if (typeof child === "string" || typeof child === "number") {
+      return {
+        type: "TEXT_ELEMENT",
+        props: { nodeValue: String(child) },
+      };
+    }
+    return child;
+  });
```

Comment:
```suggestion
  const flatChildren = children
    .flat(Infinity)
    .filter((child) => child != null && typeof child !== "boolean")
    .map((child) => {
      if (typeof child === "string" || typeof child === "number") {
        return {
          type: "TEXT_ELEMENT",
          props: { nodeValue: String(child) },
        };
      }
      return child;
    });
```
지금 방식도 가독성이 좋은 것 같아서 괜찮은 것 같지만 제 개인적인 취향은 `let`으로 선언해서 계속 재할당 하는 것보다 이렇게 체이닝해서 사용하는 걸 선호하는 것 같아요!